client.go

```diff
@@ -71,17 +78,21 @@ func (version APIVersion) String() string {
 // Client is the basic type of this package. It provides methods for
 // interaction with the API.
 type Client struct {
+	httpClient *http.Client
+
+	addresses []string
+	username  string
+	secret    string
+	userAgent string
+
+	configLock  *sync.RWMutex // Lock for config changes
```

Comment:
Githhub PR diff sucks so leaving the comment here: I think `Client.checkAPIVersion` needs the config lock too